src/ert/_c_wrappers/enkf/enkf_main.py

```diff
@@ -107,44 +201,61 @@ def _generate_parameter_files(
         iens: Realisation index
         fs: EnkfFs from which to load parameter data
     """
-    exports = {}
+    exports: Dict[str, Dict[str, float]] = {}
     for key in ens_config.getKeylistFromVarType(
-        EnkfVarType.PARAMETER + EnkfVarType.EXT_PARAMETER
+        EnkfVarType.PARAMETER + EnkfVarType.EXT_PARAMETER  # type: ignore
     ):
         node = ens_config[key]
-        enkf_node = EnkfNode(node)
-        node_id = NodeId(report_step=0, iens=iens)
+        type_ = node.getImplementationType()
 
-        if node.getUseForwardInit() and not enkf_node.has_data(fs, node_id):
+        if type_ == ErtImplType.FIELD:
+            if node.getUseForwardInit() and not fs.field_has_data(key, iens):
+                continue
+            _generate_field_parameter_file(
+                fs,
+                iens,
+                node.getFieldModelConfig(),
+                node.get_enkf_outfile(),
+                Path(run_path),
+            )
             continue
-        enkf_node.load(fs, node_id)
 
-        node_eclfile = node.get_enkf_outfile()
+        if type_ == ErtImplType.GEN_KW:
+            _generate_gen_kw_parameter_file(
+                fs,
+                iens,
+                node.getKeywordModelConfig(),
+                node.get_enkf_outfile(),
+                Path(run_path),
+                exports,
```

Comment:
Should we consider not passing the `exports` dict and instead return from `_generate_gen_kw_parameter_file` and the other `_generate...` functions?

Reply:
good point! made an issue https://github.com/equinor/ert/issues/4813